packages/amazonq/test/unit/amazonqGumby/transformationResultsHandler.test.ts

```diff
@@ -28,34 +37,80 @@ describe('DiffModel', function () {
 
             return true
         })
+        testDiffModel.parseDiff(
+            getTestResourceFilePath('resources/files/addedFile.diff'),
+            workspacePath,
+            parsedTestDescriptions.content[0],
+            1
+        )
 
-        testDiffModel.parseDiff(getTestResourceFilePath('resources/files/addedFile.diff'), workspacePath)
-
-        assert.strictEqual(testDiffModel.changes.length, 1)
-        const change = testDiffModel.changes[0]
+        assert.strictEqual(
+            testDiffModel.patchFileNodes[0].patchFilePath,
+            getTestResourceFilePath('resources/files/addedFile.diff')
+        )
+        assert(testDiffModel.patchFileNodes[0].label.includes(parsedTestDescriptions.content[0].name))
+        const change = testDiffModel.patchFileNodes[0].children[0]
 
         assert.strictEqual(change instanceof AddedChangeNode, true)
     })
 
     it('WHEN parsing a diff patch where a file was modified THEN returns an array representing the modified file', async function () {
         const testDiffModel = new DiffModel()
 
-        const workspacePath = os.tmpdir()
+        const fileAmount = 1
+        const workspaceFolder = await createTestWorkspace(fileAmount, { fileContent: '' })
 
         sinon.replace(fs, 'exists', async (path) => true)
 
         await fs.writeFile(
-            path.join(workspacePath, 'README.md'),
+            path.join(workspaceFolder.uri.fsPath, 'README.md'),
             'This guide walks you through using Gradle to build a simple Java project.'
         )
 
-        testDiffModel.parseDiff(getTestResourceFilePath('resources/files/modifiedFile.diff'), workspacePath)
+        testDiffModel.parseDiff(
+            getTestResourceFilePath('resources/files/modifiedFile.diff'),
+            workspaceFolder.uri.fsPath,
+            parsedTestDescriptions.content[0],
+            1
+        )
 
-        assert.strictEqual(testDiffModel.changes.length, 1)
-        const change = testDiffModel.changes[0]
+        assert.strictEqual(
+            testDiffModel.patchFileNodes[0].patchFilePath,
+            getTestResourceFilePath('resources/files/modifiedFile.diff')
+        )
+        assert(testDiffModel.patchFileNodes[0].label.includes(parsedTestDescriptions.content[0].name))
+        const change = testDiffModel.patchFileNodes[0].children[0]
 
         assert.strictEqual(change instanceof ModifiedChangeNode, true)
+    })
+
+    it('WHEN parsing a diff patch where diff.json is not present and a file was modified THEN returns an array representing the modified file', async function () {
+        const testDiffModel = new DiffModel() //try to see if you can reuse that test model from above
+
+        const fileAmount = 1
+        const workspaceFolder = await createTestWorkspace(fileAmount, { fileContent: '' })
 
-        await fs.delete(path.join(workspacePath, 'README.md'), { recursive: true })
+        sinon.replace(fs, 'exists', async (path) => true)
```

Comment:
What is the reason for this? Can the file be manually created instead?

Reply:
Now realizing we are just parsing a diff from the test resources directory and making sure it gets identified as the right type of node with the right label. I don't think we need to create the README.md, will fix.

Reply:
After testing, it seems we do require this information.